RandBLAS/trig_skops.hh

```diff
@@ -10,47 +10,25 @@
 #include <stdio.h>
 #include <stdexcept>
 #include <string>
+#include <sys/_types/_int64_t.h>
 #include <tuple>
 
 #include <math.h>
 #include <typeinfo>
+#include <vector>
 
 #define MAX(a, b) (((a) < (b)) ? (b) : (a))
 #define MIN(a, b) (((a) < (b)) ? (a) : (b))
 
 namespace RandBLAS {
-
     // =============================================================================
     /// WARNING: None of the following functions or overloads thereof are part of the
     /// public API
     ///
 
     // Generates a vector of Rademacher entries using the Random123 library
-    template<SignedInteger sint_t, typename RNG = r123::Philox4x32>
-    void generate_rademacher_vector_r123(sint_t* buff, uint32_t key_seed, uint32_t ctr_seed, int64_t n) {
-        RNG rng;
-
-        typename RNG::ctr_type c;
-        typename RNG::key_type key = {{key_seed}};
-
-        // Sequential loop for generating Rademacher entries
-        for (int64_t i = 0; i < n; ++i) {
-            // Set the counter for each random number
-            c[0] = ctr_seed + i;  // Ensure each counter is unique
-
-            // Generate a 2x32-bit random number using the Philox generator
-            typename RNG::ctr_type r = rng(c, key);
-
-            // Convert the random number into a float in [0, 1) using u01fixedpt
-            float rand_value = r123::u01fixedpt<float>(r.v[0]);
-
-            // Convert the float into a Rademacher entry (-1 or 1)
-            buff[i] = rand_value < 0.5 ? -1 : 1;
-        }
-    }
-
-    template<SignedInteger sint_t, typename RNG = r123::Philox4x32>
-    RandBLAS::RNGState<RNG> generate_rademacher_vector_r123(sint_t* buff, int64_t n, RandBLAS::RNGState<RNG> seed_state) {
+    template<SignedInteger sint_t = int64_t, typename RNG = DefaultRNG>
+    RNGState<RNG> generate_rademacher_vector_r123(sint_t* buff, int64_t n, RNGState<RNG> seed_state) {
```

Comment:
Actually, when it comes to templating RNGStates, please use the following structure
```
typename state_t = RNGState<DefaultRNG>
```